src/java.base/share/classes/java/lang/foreign/GroupLayout.java

```diff
@@ -73,4 +76,144 @@ public sealed interface GroupLayout extends MemoryLayout permits StructLayout, U
      */
     @Override
     GroupLayout withByteAlignment(long byteAlignment);
+
+    /**
+     * {@return a {@link Function} that can project {@linkplain MemorySegment MemorySegments} onto new instances
```

Comment:
Since a new record is returned, 'into' seems more correct. ('onto' sounds like there is an existing record instance).
```suggestion
     * {@return a {@link Function} that can project {@linkplain MemorySegment MemorySegments} into new instances
```

Reply:
"project ... onto" makes sense I think. But maybe we can reword here.